src/main/resources/com/owlcyberdefense/syslog/xsd/syslog-rfc5424.dfdl.xsd

```diff
@@ -0,0 +1,215 @@
+              <xs:pattern value="[!-~]+" />
+            </xs:restriction>
+          </xs:simpleType>
+        </xs:element>
+      </xs:sequence>
+    </xs:sequence>
+  </xs:group>
+
+  <xs:group name="rfc5424BodyGroup">
+    <xs:sequence>
+      <xs:choice>
+        <!--
+        NoStructuredData was an element with length="0" and an initiator="-"
+        but that's not portable as IBM DFDL doesn't put that into the infoset.
+        So we model this as a length 1 element instead.
+
+        <xs:element name="NoStructuredData" type="xs:string" dfdl:lengthKind="explicit" dfdl:length="1">
+          <xs:annotation>
+          <xs:appinfo source="http://www.ogf.org/dfdl/">
+            <dfdl:assert>{ . eq '-' }</dfdl:assert>
+          </xs:appinfo>
+          </xs:annotation>
+        </xs:element>
+        -->
+        <xs:sequence dfdl:initiator="-"/>
+        <xs:element name="StructuredData">
+          <xs:complexType>
+            <xs:sequence>
+              <xs:element name="SDElement" maxOccurs="unbounded">
+                <xs:complexType>
+                  <xs:sequence dfdl:initiator="[" dfdl:terminator="]" dfdl:separator="%SP;">
+                    <xs:element name="SDID">
+                      <xs:simpleType>
+                        <xs:restriction base="xs:string">
+                          <xs:minLength value="1" />
+                          <xs:maxLength value="32" />
+                          <xs:pattern value="[!-~-[= \]&quot;]]+" />
+                        </xs:restriction>
+                      </xs:simpleType>
+                    </xs:element>
+                    <xs:element name="SDParam" minOccurs="0" maxOccurs="unbounded">
+                      <xs:complexType>
+                        <xs:sequence dfdl:separator="=">
+                          <xs:element name="ParamName">
+                            <xs:simpleType>
+                              <xs:restriction base="xs:string">
+                                <xs:minLength value="1" />
+                                <xs:maxLength value="32" />
+                                <xs:pattern value="[!-~-[= \]&quot;]]+" />
+                              </xs:restriction>
+                            </xs:simpleType>
+                          </xs:element>
+                          <xs:element name="ParamValue" type="xs:string" dfdl:encoding="UTF-8" dfdl:escapeSchemeRef="sl:paramValueES" />
+                        </xs:sequence>
+                      </xs:complexType>
+                    </xs:element>
+                  </xs:sequence>
+                </xs:complexType>
+              </xs:element>
+            </xs:sequence>
+          </xs:complexType>
+        </xs:element>
+      </xs:choice>
+      <xs:element name="Message" minOccurs="0" dfdl:initiator="%SP;">
+        <xs:complexType>
+          <xs:choice>
+            <xs:element name="UTF8" dfdl:initiator="%#xEF;%#xBB;%#xBF;" type="xs:string" dfdl:encoding="UTF-8" />
```

Comment:
There is no upper bound on length. The RFC5424 spec says:
>  Syslog message size limits are dictated by the syslog transport
   mapping in use.  There is no upper limit per se.  Each transport
   mapping defines the minimum maximum required message length support,
   and the minimum maximum MUST be at least 480 octets in length.
>
>   Any transport receiver MUST be able to accept messages of up to and
   including 480 octets in length.  All transport receiver
   implementations SHOULD be able to accept messages of up to and
   including 2048 octets in length.  Transport receivers MAY receive
   messages larger than 2048 octets in length.  If a transport receiver
   receives a message with a length larger than it supports, the
   transport receiver SHOULD truncate the payload.  Alternatively, it
   MAY discard the message.
>
>   If a transport receiver truncates messages, the truncation MUST occur
   at the end of the message.  After truncation, the message MAY contain
   invalid UTF-8 encoding or invalid STRUCTURED-DATA.  The transport
   receiver MAY discard the message or MAY try to process as much as
   possible in this case.

So there are minimum lengths that must be supported, but I'm sure modern day syslog implements support significantly larger than that. Syslog messages must fit in a single UDP packet, so that is likely the real upper bound, but that's like 65k bytes.